### PR TITLE
Add `TargetDirectory` field to subscriptions

### PR DESCRIPTION
**Pull request by premun**
- Adds the DB column
- Adds the field to API controllers
- Regenerates the API C# client
- Extends the `darc` commands/pop-ups with this new field

https://github.com/dotnet/arcade-services/issues/3368

### Release Note Category
- [x] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements

### Release Note Description
Added `TargetDirectory` field to subscriptions

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/Darc/Models/PopUps/UpdateSubscriptionPopUp.cs [hunk 1]
```diff
@@ -32,7 +32,14 @@ public class UpdateSubscriptionPopUp : SubscriptionPopUp
             header: [
                 new Line($"Use this form to update the values of subscription '{subscription.Id}'.", true),
                 new Line($"Note that if you are setting 'Is batchable' to true you need to remove all Merge Policies.", true),
-                new Line()
+                Line.Empty,
+                new("Source and target directories only apply to source-enabled subscription (VMR code flow subscriptions).", true),
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Darc/Darc/Models/PopUps/UpdateSubscriptionPopUp.cs:36]: nit
```suggestion
                new("Source and target directories only apply to source-enabled subscriptions (VMR code flow subscriptions).", true),
```

File: src/Microsoft.DotNet.Darc/Darc/Models/PopUps/UpdateSubscriptionPopUp.cs [hunk 2]
```diff
@@ -32,7 +32,14 @@ public class UpdateSubscriptionPopUp : SubscriptionPopUp
...
+                new("They define which directory of the VMR (under src/) are the sources synchronized with.", true),
+                new("Only one of those needs to be set based on whether the source or the target repo is the VMR.", true),
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Darc/Darc/Models/PopUps/UpdateSubscriptionPopUp.cs:38]:
```suggestion
                new("Only one of those can be set based on whether the source or the target repo is the VMR.", true),
```

Comment (premun) [reply src/Microsoft.DotNet.Darc/Darc/Models/PopUps/UpdateSubscriptionPopUp.cs:38]: I think needs to, because one must be set

Comment (dkurepa) [reply src/Microsoft.DotNet.Darc/Darc/Models/PopUps/UpdateSubscriptionPopUp.cs:38]: when I'm reading this, `needs to be` means that 1 needs to be set, but you can also set the other one. Perhaps `must be` set is better?

Comment (premun) [reply src/Microsoft.DotNet.Darc/Darc/Models/PopUps/UpdateSubscriptionPopUp.cs:38]:
```suggestion
                new("Only one of those must be set based on whether the source or the target repo is the VMR.", true),
```

Comment (premun) [reply src/Microsoft.DotNet.Darc/Darc/Models/PopUps/UpdateSubscriptionPopUp.cs]: I put `must` but `needs to` vs `must` is almost the same?

**Comment by oleksandr-didyk** [comment]
Rolled out by https://github.com/dotnet/arcade-services/pull/3476

**Pull request merged**